src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -764,16 +765,24 @@ ExecuteSingleTask(QueryDesc *queryDesc, Task *task,
 		ShardPlacement *taskPlacement = (ShardPlacement *) lfirst(taskPlacementCell);
 		bool queryOK = false;
 		bool failOnError = false;
+		PGconn *connection = NULL;
 		int64 currentAffectedTupleCount = 0;
-		PGconn *connection = GetConnectionForPlacement(taskPlacement,
-													   isModificationQuery);
-
-		if (connection == NULL)
+		int connectionFlags = SESSION_LIFESPAN;
+		MultiConnection *multiConnection = GetNodeConnection(connectionFlags,
+															 taskPlacement->nodeName,
+															 taskPlacement->nodePort);
+		if (multiConnection == NULL)
```

Comment:
Node that you *always* get back a multiConnection, so this test isn't really meaningful.  If you want to actually test the health you'd need to copy the test from GetOrEstablishConnection().